pkgs/by-name/so/somo/package.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  installShellFiles,
+  versionCheckHook,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "somo";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "theopfr";
+    repo = "somo";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-HUTaBSy3FemAQH1aKZYTJnUWiq0bU/H6c5Gz3yamPiA=";
+  };
+
+  cargoHash = "sha256-e3NrEfbWz6h9q4TJnn8jnRmMJbeaEc4Yo3hFlaRLzzQ=";
+
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    installShellCompletion --cmd somo \
+      --bash <("$out/bin/somo" generate-completions bash) \
+      --zsh <("$out/bin/somo" generate-completions zsh) \
+      --fish <("$out/bin/somo" generate-completions fish)
+  '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  doInstallCheck = true;
+  versionCheckProgram = "${placeholder "out"}/bin/somo";
```

Comment:
```suggestion
```

This should be the default.

Reply:
Usually we don't set this parameter if the default works, so I would prefer to remove this line. Regarding robustness, in #412768 the default has been changed to `meta.mainProgram`.

Reply:
>https://github.com/NixOS/nixpkgs/pull/412768 the default has been changed to meta.mainProgram.

Oh, I didn't notice the pull request and discussion. Thank you for sharing them. Then, this line should use the default one.